aws/credentials/credentials.go

```diff
@@ -1,61 +1,64 @@
 package credentials
 
 import (
-	"net/http"
+	"context"
+	"log"
 	"time"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/credentials"
-	"github.com/aws/aws-sdk-go/aws/credentials/ec2rolecreds"
-	"github.com/aws/aws-sdk-go/aws/ec2metadata"
-	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/transport/http"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/credentials/ec2rolecreds"
+	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
 )
 
 type AWSCredentials interface {
-	Get() (*credentials.Credentials, error)
+	Get() (*aws.Credentials, error)
 }
 
 type awsCred struct {
-	creds *credentials.Credentials
+	creds *aws.Credentials
```

Comment:
Store the CredentialsProvider interface as well as part of this struct and add a new function `GetCredentialsProvider` to the above interface.

Reply:
Addressed in this [commit](https://github.com/libopenstorage/secrets/pull/76/commits/6a1cf46cb0faa8455010bce92100ac7ce4d99d69)